### PR TITLE
View model switch

### PR DESCRIPTION
**Pull request by Kethsar**
Transitioned methods for Departments and Opportunities to use View Models
Incorporated nifty JavaScript for adding majors to Departments and Opportunities

### CODE REVIEW

**Comment by Kethsar** [comment]
I need to properly integrate my changes with the new controllers.

**Comment by Kethsar** [comment]
Finished transferring logic
Also fixed relationships between some objects

File: Coop_Listing_Site/Coop_Listing_Site/Views/Department/Edit.cshtml [hunk 1]
```diff
@@ -28,19 +28,112 @@
...
             <div class="col-md-offset-2 col-md-10">
                 <input type="submit" value="Save" class="btn btn-primary" />
-                @Html.ActionLink("Back to List", "", "Department", new { id = Model.DepartmentID }, new { @class = "btn btn-primary" })
+                @Html.ActionLink("Back to List", "", "Department", null, new { @class = "btn btn-primary" })
             </div>
         </div>
     </div>
 }
 
+<script type="text/javascript">
```
Comment (Chef-Code) [inline Coop_Listing_Site/Coop_Listing_Site/Views/Department/Edit.cshtml:66]: @Kethsar I'm stealing this method for my recipe-builder app.  This is exactly what I've been needing.

File: Coop_Listing_Site/Coop_Listing_Site/Views/Coop/AddOpportunity.cshtml [hunk 2]
```diff
@@ -166,13 +166,34 @@
...
+                @Html.DropDownList("DepartmentIDs", "Select One")
             </div>
         </div>
         <div class="form-group">
-            @Html.Label("Major", htmlAttributes: new { @class = "control-label col-md-2" })
+            @Html.Label("Majors", htmlAttributes: new { @class = "control-label col-md-2" })
             <div class="col-md-10">
-                @Html.DropDownList("MajorIDs", "Select One")
+                <select id="Majors" on>
+                    <option id="top-opt" value>Choose a Department first</option>
```
Comment (Chef-Code) [inline Coop_Listing_Site/Coop_Listing_Site/Views/Coop/AddOpportunity.cshtml:176]: @Kethsar  
Subject of reference: &lt;option id="top-opt" value&gt; 
Question in regards: Why is value not instantiated? Is this satisfied through the selection of the option?

File: Coop_Listing_Site/Coop_Listing_Site/Views/Coop/AddOpportunity.cshtml [hunk 3]
```diff
@@ -187,4 +208,126 @@
...
+            $.post("/coop/getmajorsfordepartment", { id: id }, function (majors) {
+                for (var i = 0; i < majors.length; i++) {
+                    for (var j = 0; j < curMajors.length; j++) {
+                        if (curMajors[j].value == majors[i].MajorID)
+                            inList = true;
+                    }
+
+                    if (!inList) {
+                        var option = document.createElement("option");
+                        option.value = majors[i].MajorID;
```
Comment (Chef-Code) [inline Coop_Listing_Site/Coop_Listing_Site/Views/Coop/AddOpportunity.cshtml:308]: @Kethsar 
Is it populated from this line of code?

File: Coop_Listing_Site/Coop_Listing_Site/Views/ControlPanel/AddDepartment.cshtml [hunk 4]
```diff
@@ -32,6 +32,24 @@
...
+            @Html.Label("Majors In Department", htmlAttributes: new { @class = "control-label col-md-2" })
+            <div class="col-md-10">
+                <ol style="list-style-type: none; padding-left: 0px;">
+                    @if (Model != null && Model.Majors.Count > 0)
+                    {
+                        foreach (var major in Model.Majors)
+                        {
+                            <div id="@major.MajorID">
+                                <button type="button" onclick="removeMajor(@major.MajorID);">X</button> <li style="display:inline-block;" value="@major.MajorID">@major.MajorName</li>
+                                <br />
```
Comment (Chef-Code) [inline Coop_Listing_Site/Coop_Listing_Site/Views/ControlPanel/AddDepartment.cshtml:44]: @Kethsar  This is nicely done

File: Coop_Listing_Site/Coop_Listing_Site/Views/ControlPanel/AddDepartment.cshtml [hunk 5]
```diff
@@ -42,3 +60,78 @@
 <div>
     @Html.ActionLink("Back to List", "Index")@*can't seem to make this a button*@
 </div>
```
Comment (Chef-Code) [inline Coop_Listing_Site/Coop_Listing_Site/Views/ControlPanel/AddDepartment.cshtml:62]: > `@Html.ActionLink("Back to List", "Index")@*can't seem to make this a button*@` 

Would this work??
`<button value='@Html.ActionLink("Back to List", "Index")'></button>`

**Comment by Chef-Code** [comment]
From what I can tell this is good to go.

**Comment by Chef-Code** [comment]
@Kethsar I'll let you delete the branch.  I think you don't care, but I am not 100% certain

**Comment by Kethsar** [comment]
@Chef-Code in regards to the option element with an empty value, it's to help with the if statement on line 296.
 I grab the id from the value of the option. If there is no value attribute, it grabs the text inside the option instead. The empty value makes it so it grabs an empty string, which JavaScript implicitly ties to 'false' when used in an if statement the way I do.

**Pull request merged**